packages/flutter_test/lib/src/binding.dart

```diff
@@ -507,6 +507,12 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   }
 
   Future<Null> _runTestBody(Future<Null> testBody(), VoidCallback invariantTester) async {
+    // Delay this function by a microtask.
+    // Otherwise it will open a scope immediately, which is then open when
+    // the `asyncBarrier` is invoked. The `asyncBarrier` is immediately
+    // following the call to `testZone.runBinary(_runTestBody)`, so delaying
+    // by one microtask is enough to ensure that the timing is correct.
+    await new Future<Null>.microtask(() {});
```

Comment:
I don't really understand this comment. Can you extend it to more fully explain the sequence of events?

Reply:
I think what I'm having trouble with is the question of what the "open scope" that this comment is referring to is. Where do we open a scope which delaying the code here helps with? This microtask doesn't interact with the async barrier, since the async barrier isn't called until later within this method (synchronously) anyway. Indeed, the async barrier can't possibly be what drives the microtasks (as per the last sentence of the comment here) since it isn't invoked until after the microtasks are invoked, since it's called synchronously by this method.